test/automl/test_classification.py

```diff
@@ -277,12 +277,15 @@ def test_sparse_matrix_xgboost(self):
         import subprocess
         import sys
 
+        current_xgboost_version = xgb.__version__
         subprocess.check_call([sys.executable, "-m", "pip", "install", "xgboost==1.3.3", "--user"])
         automl = AutoML()
         automl.fit(X_train=X_train, y_train=y_train, **automl_settings)
         print(automl.feature_names_in_)
         print(automl.feature_importances_)
-        subprocess.check_call([sys.executable, "-m", "pip", "install", "-U", "xgboost", "--user"])
```

Comment:
This is the root cause of the difference in XGBoost versions between the executor and driver of spark session.

Reply:
Amazing bug fix, this could also explain why my xgb is always the latest version after pytest. We should avoiding modifying environment in this way.